### PR TITLE
Return rejected promise on error [3.x]

### PR DESCRIPTION
**Pull request by mitsos1os**
Backports #1790  to 3.X branch of juggler 

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-datasource-juggler) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](https://loopback.io/doc/en/contrib/style-guide-es6.html)
- [x] Commit messages are following our [guidelines](https://loopback.io/doc/en/contrib/git-commit-messages.html)

### CODE REVIEW

File: test/manipulation.test.js [hunk 1]
```diff
@@ -1552,6 +1567,41 @@ describe('manipulation', function() {
         done();
       });
     });
+
+    it('correctly coerces the PK value', async () => {
```
Comment (bajtos) [inline test/manipulation.test.js]: AFAICT, this test was not present in https://github.com/strongloop/loopback-datasource-juggler/pull/1790. I am not entirely opposed to add it, but would like to understand your motivation & reasoning first.

Comment (mitsos1os) [reply test/manipulation.test.js]: @bajtos I am sorry, I cherry picked the commit and brought over the test.
You are right this is not related, and I will just omit the test for 3.X  branch

**Comment by bajtos** [comment]
@slnode ok to test

**Comment by dhmlau** [comment]
FYI - I've removed the required checks for Node.js 6.x  for the 3.x branch.

**Comment by bajtos** [comment]
**[cis-jenkins] downstream: loopback-connector-couchdb2@master — Failed! (315545d)**

```
17:31:03   1) CouchDB2 imported features
17:31:03        manipulation
17:31:03          create
17:31:03            should return rejected promise when model initialization failed:
17:31:03      TypeError: Person.create(...).should.be.rejectedWith is not a function
17:31:03       at Context.it (node_modules/loopback-datasource-juggler/test/manipulation.test.js:124:13)
```

This test failure looks legitimate to me. I suspect our CouchDB2 connector is using outdated version of shouldjs (v7.1.1, see [here](https://github.com/strongloop/loopback-connector-couchdb2/blob/master/package.json#L47)), while juggler uses v8.x in the 3.x branch (see [here](https://github.com/strongloop/loopback-datasource-juggler/blob/6ed60653eb67d975c2b93fdda93db89318a05e54/package.json#L47)).

I think we should update the CouchDB2 connector to use the same version of should as loopback-datasource-juggler does.

@jannyHou WDYT?

**[cis-jenkins] downstream: loopback-connector-postgresql@master — Failed! (315545d)**

AFAICT, this test failure is unrelated - I see a lot of timeout errors.

**Comment by bajtos** [comment]
> I think we should update the CouchDB2 connector to use the same version of should as loopback-datasource-juggler does.

See https://github.com/strongloop/loopback-connector-couchdb2/pull/74.

Let's wait for that pull request to be landed, then I'll trigger a new run of downstream CI builds.

**Comment by bajtos** [comment]
@slnode test please

**Comment by bajtos** [comment]
@slnode test please

**Comment by bajtos** [comment]
Landed, thank you @mitsos1os for back-porting the fix to 3.x 🙇

**Pull request merged**